plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java

```diff
@@ -778,6 +791,49 @@ public Object[][] unsupportedTimestampDataProvider()
         };
     }
 
+    @Test(dataProvider = "sessionZonesDataProvider")
+    public void testClickHouseDateTimeWithTimeZone(ZoneId sessionZone)
+    {
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(sessionZone.getId()))
+                .build();
+
+        dateTimeWithTimeZoneTest(clickhouseDateTimeInputTypeFactory("datetime"), clickhouseDateTimeInputLiteralFactory(), trinoTimestampExpectedLiteralFactory())
+                .execute(getQueryRunner(), session, clickhouseCreateAndInsert("tpch.datetime_tz"));
+    }
+
+    private SqlDataTypeTest dateTimeWithTimeZoneTest(Function<ZoneId, String> inputTypeFactory, BiFunction<LocalDateTime, ZoneId, String> inputLiteralFactory, BiFunction<LocalDateTime, ZoneId, String> expectedLiteralFactory)
+    {
+        ZoneId utc = ZoneId.of("UTC");
+        SqlDataTypeTest tests = SqlDataTypeTest.create()
+                .addRoundTrip(format("Nullable(%s)", inputTypeFactory.apply(utc)), "NULL", TIMESTAMP_TZ_SECONDS, "CAST(NULL AS TIMESTAMP(0) WITH TIME ZONE)");
+
+        // Since ClickHouse datetime(timezone) does not support values before epoch, we do not test this here.
+        tests.addRoundTrip(inputTypeFactory.apply(utc), "0", TIMESTAMP_TZ_SECONDS, expectedLiteralFactory.apply(epoch.atZone(utc).withZoneSameInstant(utc).toLocalDateTime(), utc))
+                .addRoundTrip(inputTypeFactory.apply(utc), inputLiteralFactory.apply(epoch, utc), TIMESTAMP_TZ_SECONDS, expectedLiteralFactory.apply(epoch.atZone(utc).withZoneSameInstant(utc).toLocalDateTime(), utc))
```

Comment:
the literal factory is useful for reuse but IMO for tests it's easier to "see" the literals as strings.

Also avoids the issue that any code in the literal factory can affect the test. This is the main reason we moved away from old `DataTypeTest` to `SqlDataTypeTest`.

Reply:
(Also, please squash the commits when you apply this review comment since I don't expect there to be more comments).

Reply:
Applied and squashed commits, I think you will need to do a final round of REVIEW :)